contracts/libs/Actions.sol

```diff
@@ -62,6 +98,46 @@ library Actions {
         uint256 amount;
     }
 
+    /**
+     * @notice Parses the passed in action argmuents to get the argmuents for a mint action
+     * @param _args The general action arguments structure
+     * @return The arguments for a mint action
+     */
+    function _parseMintArgs(ActionArgs memory _args) internal returns (MintArgs memory) {
+        require(_args.actionType == ActionType.MintShortOption, "Actions: can only parse arguments for mint actions");
+        require(_args.owner != address(0), "Actions: cannot mint from an invalid account");
+
+        return
+            MintArgs({
+                owner: _args.owner,
+                vaultId: _args.vaultId,
+                to: _args.sender,
```

Comment:
what does `to: _args.sender` refer to ? so in this case the minted short option will be stored in the `vault.vaultId` and the tokens will be minted to the `sender` and not the `owner` right ? the sender can be operator also.

What happen in the scenario where an operator mint token for a vault, and he get the tokens, and then the vault owner want to do some operation with the short option amount, but he doesn't hold them, but still the short options are stored in his vault ?

Reply:
yeah, that's why its important for the end user to be careful about who they authorize the vault to. The checks on sender is vault operator should all happen in the controller.

Reply:
tbh the vault operator can also be the person who deposited collateral. So it works both ways

Reply:
maybe initially we can whitelist those addresses that can be vault operators? @antoncoding

Reply:
yeah my point is I think minted option should always be minted to the vault owner. Operator will just operate vault, but will not get ownership of minted option.

Reply:
Either way it's not a concern here, all the scenario you guys discuss will be handle in the Controller (check MintArg.to == owner or whatever), here we're just parsing the arguments from general Arg type to MintArgs.

Reply:
@haythem96 any final thoughts or can I resolve this?

Reply:
@antoncoding can u elaborate more ? my concern is regarding minting short options to operator address, will that be possible or all minted short will be sent to owner ?

Reply:
For now, in this library, we only convert general Args struct into `MintArgs`(parseMintArgs function). it will only does basic checks around invalid address, and the real access control logic should be done in the controller. 
```javascript
// Controller.sol
function mintShortOtokens(MintArgs args) {
    require(msg.sender == args.owner || isOperator[args.owner][msg.sender], "Tx sender must be owner or operator.")
    require(args.to == owner || args.to == msg.sender, "Cannot mint to arbitrary accounts") // if needed
}
```

I think those logics belong to the process of "minting tokens", instead of `parsing arguments`, so I would want to put these in the `mintShortOtokens` instead of `parseMintArgs`

Reply:
yeah, I'm not talking about the checks, so in those two requires, the operator can can call mint action, and get the minted options in his balance, what's ur thought re that ?!

Reply:
would love ur thoughts @zkoticha @aleone .

Reply:
@haythem96 those 2 requirements are in Controller, so we should still be able to merge this now.

And if we want to empower a wrapper contract, I think it's essential to let the wrapper decide where to mint tokens to. If the wrapper can only mint token to the owner, it won't be able to play something like "flashloan", because after it mints the token to the owner's account, it can't move those tokens until the owner approve again.